### ISSUE TITLE
About lr

### ISSUE THREAD

**Issue by wangyu1213888** [haliphinx/TCE-RVOS#1]
Hi, I have a question about the code in the engine, why do you  add the learning rate update after each losses.backward(). And I wonder what the difference is between dist_train_test_ytvos_new.sh and dist_train_test_ytvos.sh. Which one should I use if I want to achieve 59.6 with ResNet-50
`optimizer.step()`
 `if lr_scheduler is not None:`
     `       lr_scheduler.step()`

**Comment by haliphinx** [comment]
Hi, that is used for cyclic learning rate training. However, there is a bug in the main.py which seems not distinguishing the original learning rate with the cyclic_lr. We just fixed that part as: [this part](https://github.com/haliphinx/TCE-RVOS/blob/0308b7468400b89245d96a0448537c3ee5c53a19/main.py#L255C9-L257C32).

Cyclic learning rate updates the lr after each backward pass and the original lr strategy only decreases after the 3rd and 5th epoch. That is why there is a "learning rate update" at that place. This cyclic learning rate is only used for testing, which is not used in the final.

Please refer to [training script](https://github.com/haliphinx/TCE-RVOS/blob/fb4db36dc63ec864d8ebb9edb7750fef79509385/scripts/dist_train_test_ytvos_new.sh) for how to start the training.

**Comment by wangyu1213888** [comment]
Thank you very much for your answer
